### ISSUE TITLE
Could not create change - changing role resource

### ISSUE THREAD

**Issue by magnuslarsen** [elastic/puppet-elasticsearch#1013]
* Module version: Elasticsearch 6.3.3
* Puppet version: 5.5.7
* OS and version: Ubuntu 18.04

## Bug description
When changing multiple role resources, only the first resource gets updates, rest gets an error:
```
Info: Applying configuration version 'puppetmaster-branchname-uid'
Notice: /Stage[main]/Profile::Clm::Elasticsearch/Elasticsearch::Role[logstash_role1]/Elasticsearch_role[logstash_role1]/privileges: privileges changed {
  'indices' => [
    {
      'names' => ['path0_*'],
      'privileges' => ['write', 'create', 'create_index']
    }]
} to {
  'indices' => {_l
    'names' => ['path1_*'],
    'privileges' => ['write', 'create', 'create_index']
  }
}
Info: Computing checksum on file /etc/elasticsearch/roles.yml
Error: Could not convert change 'privileges' to string: no block given (yield)
Error: Could not convert change 'privileges' to string: no block given (yield)
Error: /Stage[main]/Profile::Clm::Elasticsearch/Elasticsearch::Role[logstash_role2]/Elasticsearch_role[logstash_role2]/privileges: could not create change error message for privileges
Error: /Stage[main]/Profile::Clm::Elasticsearch/Elasticsearch::Role[logstash_role2]/Elasticsearch_role[logstash_role2]: Could not evaluate: no block given (yield)
```
If you apply the the manifest again, `logstash_role2` succeeds, but `logstash_role3` fails... and so on, until every resource has been applied..

Puppet resource declaration (snippet):
```
class profile::clm::elasticsearch {
  elasticsearch::role { 'logstash_role1':
    mappings   => ['CN=logstash1,DC=domain,DC=tld'],
    privileges => {
      'indices' => {
        'names'      => ['path1_*'],
        'privileges' => ['write', 'create', 'create_index'],
      }
    }
  }

  elasticsearch::role { 'logstash_role2':
    mappings   => ['CN=logstash2,DC=domain,DC=tld'],
    privileges => {
      'indices' => {
        'names'      => ['path2_*'],
        'privileges' => ['write', 'create', 'create_index'],
      }
    }
  }
}
```

**Comment by tylerjl** [comment]
Thank you for the report, @magnuslarsen. Could you reproduce the issue using `--debug` or `--trace` when performing the puppet run? Those flags might show more of the error log to help track down where in the code it's occurring.

**Comment by magnuslarsen** [comment]
Absolutely. Please see this link:
https://pastebin.com/cEjxs9xB

I took everything relevant to the elasticsearch manifest.  Unfortunately it doesn't seem to point at a lot.

I created two changes this time around:
FROM:
```
  elasticsearch::role { 'logstash_role1':
    mappings   => ['CN=logstash1,DC=domain,DC=tld'],
    privileges => {
      'indices' => {
        'names'      => ['path1-new_*'],
        'privileges' => ['write', 'create', 'create_index'],
      }
    }
  }

  elasticsearch::role { 'logstash_role2':
    mappings   => ['CN=logstash2,DC=domain,DC=tld'],
    privileges => {
      'indices' => {
        'names'      => ['path2-new_*'],
        'privileges' => ['write', 'create', 'create_index'],
      }
    }
  }
```
TO:
```
  elasticsearch::role { 'logstash_role1':
    mappings   => ['CN=logstash1,DC=domain,DC=tld'],
    privileges => {
      'indices' => {
        'names'      => ['path1-new1_*'],
        'privileges' => ['write', 'create', 'create_index'],
      }
    }
  }

  elasticsearch::role { 'logstash_role2':
    mappings   => ['CN=logstash2,DC=domain,DC=tld'],
    privileges => {
      'indices' => {
        'names'      => ['path2-new1_*'],
        'privileges' => ['write', 'create', 'create_index'],
      }
    }
  }
```

**Comment by tylerjl** [comment]
Hmm, looks like the ruby stack trace piece might've gotten swallowed, I think we can get that to bubble up by running with `--trace` instead of `--debug`, could you perhaps run with that flag and let me know what it spits back? Thank you for helping with debugging!

**Comment by magnuslarsen** [comment]
Here you go: https://pastebin.com/pd8P5V6t

**Comment by tylerjl** [comment]
Thanks again @magnuslarsen, that stack trace actually pinpointed the issue. I'll see about coming up with a fix :+1:

**Comment by magnuslarsen** [comment]
Great - glad i could help!

**Comment by tylerjl** [comment]
@magnuslarsen although I _think_ I've got a fix in place, the problem is a little hard to reproduce in my test cases - when you have a moment, do you think you could test the following changes to see if they resolve the issue in your environment?

https://github.com/elastic/puppet-elasticsearch/compare/master...tylerjl:hash-each-pair-return?expand=1

**Comment by magnuslarsen** [comment]
I will test monday, and report back to you
Thanks for the help!

**Comment by magnuslarsen** [comment]
I made two changes (just like the previous times) - both got updated this time, without an error!
So.. can confirm it works :+1:

**Comment by tylerjl** [comment]
Fixed in #1090
